src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -186,19 +231,21 @@ export class WorkspaceSavedObjectsClientWrapper {
       objects: Array<SavedObjectsBulkUpdateObject<T>>,
       options?: SavedObjectsBulkUpdateOptions
     ): Promise<SavedObjectsBulkUpdateResponse<T>> => {
-      const workspaceIds = objects.reduce<string[]>((acc, cur) => {
-        if (this.isRelatedToWorkspace(cur.type)) {
-          acc.push(cur.id);
+      for (const object of objects) {
+        const workspacePermitted = await this.validateWorkspaceAttributesPermitted(
+          object.attributes,
+          wrapperOptions.request,
+          WorkspacePermissionMode.Management
+        );
+
+        if (!workspacePermitted) {
+          await this.validateSingleObjectPermissions(
+            object.id,
+            object.type,
+            wrapperOptions.request,
+            [WorkspacePermissionMode.Management, WorkspacePermissionMode.Write]
```

Comment:
Do we need to check WorkspacePermissionMode.LibraryWrite when `object.type` equal WORKSPACE?